tests/test_job.py

```diff
@@ -238,6 +238,13 @@ def test_deepcopy(self):
         copied_job.sp.a = 3
         assert copied_job in self.project
 
+    def test_project(self):
+        job = self.project.open_job({"a": 0})
+        assert job.project.path == self._tmp_pr
```

Comment:
Let's test a few more invariants.
```suggestion
        assert isinstance(job.project, signac.Project)
        assert job in job.project
        assert job.project.path == self._tmp_pr
```